### PR TITLE
(maint) Add ability to generate source artifacts for rpm/deb

### PR DESCRIPTION
**Pull request by underscorgan**
Sometimes it's useful to be able to generate .src.rpm and
.dsc/changes/etc. This adds the ability to do that.

### CODE REVIEW

**Comment by underscorgan** [comment]
Still need to add docs and tests.

File: lib/vanagon/platform.rb [hunk 1]
```diff
@@ -60,6 +60,9 @@ class Platform
     # Stores the local path where retrieved artifacts will be saved.
     attr_accessor :output_dir
 
+    # Stores the local path where source artifacts will be saved.
+    attr_accessor :source_dir
```
Comment (shrug) [inline lib/vanagon/platform.rb]: Hmm.... I kinda think this is a little bit confusing, too easy to mix up with build sources. maybe source_output_dir? or something a bit more descriptive?

Comment (underscorgan) [inline lib/vanagon/platform.rb]: ah, good point. will fix

File: lib/vanagon/platform/rpm.rb [hunk 2]
```diff
@@ -40,6 +49,13 @@ def output_dir(target_repo = "products")
         super
       end
 
+      # Method to derive the directory for source artifacts
+      #
+      # @param target_repo [String] repo the source artifacts are targeting
+      def source_dir(target_repo = "products")
+        @source_dir ||= File.join(@os_name, @os_version, target_repo, 'SRPMS')
```
Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: Do we want `target_repo` to come first, since we're changing paths?

File: lib/vanagon/platform/rpm.rb [hunk 3]
```diff
@@ -40,6 +49,13 @@ def output_dir(target_repo = "products")
         super
       end
 
+      # Method to derive the directory for source artifacts
+      #
+      # @param target_repo [String] repo the source artifacts are targeting
+      def source_output_dir(target_repo = "products")
+        @source_output_dir ||= File.join(@os_name, @os_version, target_repo, 'SRPMS')
```
Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: Do we want `target_repo` first, since we're changing paths?

File: lib/vanagon/platform/rpm.rb (hunk 2 above)
Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: No, this is happening before those vanagon changes. This should get updated when those changes go in.

File: lib/vanagon/platform/rpm.rb (hunk 3 above)
Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: Nope, those changes haven't been made yet, this will need to get updated when that happens.

Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: I'm just literally working on those changes now.

Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: But I need this to build sources for things that haven't moved yet. This is going to need to go in first.

Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: kk

Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: how are those changes going to know whether or not the order should change? not everything is moving

Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: I'm gonna check if the target repo is PC1.

Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: that's going to break PE things

Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: What do those paths look like?

Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: most of those don't have a target repo set and for RPMs that defaults to target repo of 'products' (so, like http://builds.puppetlabs.lan/puppet-enterprise-vanagon/pe-activemq-2016.4.5.14.3-3/artifacts/el/7/products/x86_64/) and debs defaults to no target repo (like http://builds.puppetlabs.lan/puppet-enterprise-vanagon/pe-activemq-2016.4.5.14.3-3/artifacts/deb/precise/)

Comment (mwaggett) [inline lib/vanagon/platform/rpm.rb]: Do we want to change those paths too though? I feel like we should..

Comment (underscorgan) [inline lib/vanagon/platform/rpm.rb]: I don't know. If we do we definitely need the change to be coordinated, the paths are for sure hardcoded in enterprise-dist.

**Pull request merged**